sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -697,7 +697,12 @@ class CodegenContext {
           }
         """
       s"${addNewFunction(compareFunc, funcCode)}($c1, $c2)"
-    case other if other.isInstanceOf[AtomicType] => s"$c1.compare($c2)"
+    case other if other.isInstanceOf[AtomicType] =>
+      if (s"$c1".getClass.getMethods.map(_.getName).filter(_ matches "compare").size == 1) {
```

Comment:
You just try to get class from a `String` object.

Reply:
@viirya i was putting a note to that effect :-) This would not work, but trying out something, especially the consistency betn local and jenkins test behavior.I needed "other" to work.Anyhow, mainly looking to check if atomictypes conv/long/String can happen by different attempts.